### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by RobertMrowiec**
# Tasks:
- create myprofile integration tests
- create add-auction integration tests
- make few fixes in code like order etc
- change fetch to ember data where it's possible

# Testing scenario:
- ember test --server 
- select every integration tests
- check how it finishes successfully

# Preview
image

### CODE REVIEW

File: tests/acceptance/add-auction-test.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	})
+
+	test('should add auction > redirect to auctions list > render this auction', async function(assert) {
+		await visit('/addAuction')
+
+		await fillIn('[data-test-auction-brand]', 'test brand')
+		await fillIn('[data-test-auction-model]', 'test model')
+		await fillIn('[data-test-auction-horsePower]', 10)
+		await fillIn('[data-test-auction-maxTorque]', 200)
+		await fillIn('[data-test-auction-assembledAt]', new Date())
```
Comment (Rxbsxn) [inline tests/acceptance/add-auction-test.js]: http://ember-cli-page-object.js.org/docs/v1.15.x/

File: tests/integration/components/add-auction-test.js [hunk 2]
```diff
@@ -1,26 +1,55 @@
...
+	test('should add auction', async function(assert) {
+		await authenticateSession(credentials)
+		await render(hbs`<Add-auction/>`)
 
-    assert.equal(this.element.textContent.trim(), '');
+		await fillIn('[data-test-auction-brand]', 'test brand')
+		await fillIn('[data-test-auction-model]', 'test model')
+		await fillIn('[data-test-auction-horsePower]', 10)
+		await fillIn('[data-test-auction-maxTorque]', 200)
+		await fillIn('[data-test-auction-assembledAt]', new Date())
```
Comment (Rxbsxn) [inline tests/integration/components/add-auction-test.js]: page objecgt

File: tests/integration/components/myprofile-test.js [hunk 3]
```diff
@@ -1,33 +1,82 @@
...
+	})
 
-		assert.dom('[data-test-input-email]').hasText('integrationtest@herecars.com')
-		assert.dom('[data-test-input-password]').hasAnyText()
-		assert.dom('[data-test-input-name]').hasText('test name')
+	test('should change user email', async function(assert) {
+		await authenticateSession(this.user)
+		await render(hbs`<Myprofile/>`)
+		await click('[data-test-myprofile-email]')
+		await fillIn('[data-test-myprofile-new]', 'anothertest@herecars.com')
```
Comment (Rxbsxn) [inline tests/integration/components/myprofile-test.js]: page object

File: tests/integration/components/myprofile-test.js [hunk 4]
```diff
@@ -1,33 +1,82 @@
...
+
+		assert.dom('#previous').hasValue('test name')
+		assert.dom('#new').exists()
+	})
+
+	test('should change user nickname', async function(assert) {
+		await authenticateSession(this.user)
+		await render(hbs`<Myprofile/>`)
+		await click('[data-test-myprofile-nickname]')
+		await fillIn('[data-test-myprofile-new]', 'another name')
```
Comment (Rxbsxn) [inline tests/integration/components/myprofile-test.js]: page object

File: app/components/add-auction.js [hunk 5]
```diff
@@ -29,45 +32,25 @@ export default Component.extend({
...
-				this.toast.set('description', 'Auction created')
-				this.get('router').transitionTo('/auction-list')
-				return later(() => this.toast.set('correctCredentials', null), 2000)
-			})
+			return this.store.createRecord('auction', { ...this.auction, car: carId })
+				.save()
+				.then(() => {
+					this.fetchHelper.checkStatusEmberData()
+					this.toast.set('correct', true)
+					this.toast.set('description', 'Auction created')
```
Comment (Rxbsxn) [inline app/components/add-auction.js]: extract to a separated method

File: app/services/auth.js [hunk 6]
```diff
@@ -9,28 +9,28 @@ export default Service.extend({
 
 	async login(credentials) {
 		if (this.get('session').isAuthenticated) {
-			this._success()
+			this._notification()
```
Comment (Rxbsxn) [inline app/services/auth.js]: Maybe there's another way to name this variable -`afterChange`/`onSuccessNotification`?

File: app/services/auth.js [hunk 7]
```diff
@@ -9,28 +9,28 @@ export default Service.extend({
...
 		} else {
 			await this.get('session').authenticate('authenticator:jwt', credentials)
-				.then(() => this._success())
-				.catch(() => this._fail())
+				.then(() => this._notification())
+				.catch(() => this._notification(false))
```
Comment (Rxbsxn) [inline app/services/auth.js]: I'd create another method for error cases - `errorCallback`/ `showError`

Comment (RobertMrowiec) [reply app/services/auth.js]: sure thing

File: tests/acceptance/add-auction-test.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+import { setupApplicationTest } from 'ember-qunit'
+import { startMirage } from 'car-bidember/initializers/ember-cli-mirage'
+import {
+	create,
+	visitable,
+	fillable,
+	clickable
+} from 'ember-cli-page-object'
+
+const page = create({
```
Comment (Rxbsxn) [inline tests/acceptance/add-auction-test.js]: Move to separated file `auction` in `page-object` directory

File: tests/acceptance/login-test.js [hunk 9]
```diff
@@ -1,8 +1,25 @@
...
 import { authenticateSession } from 'ember-simple-auth/test-support'
 import { startMirage } from 'car-bidember/initializers/ember-cli-mirage'
+import {
+	create,
+	fillable,
+	clickable,
+	visitable
+} from 'ember-cli-page-object'
+
+const page = create({
```
Comment (Rxbsxn) [inline tests/acceptance/login-test.js]: same here

File: tests/integration/components/add-auction-test.js [hunk 10]
```diff
@@ -1,26 +1,74 @@
...
+import hbs from 'htmlbars-inline-precompile'
+import { authenticateSession } from 'ember-simple-auth/test-support'
+import { startMirage } from 'car-bidember/initializers/ember-cli-mirage'
+import {
+	create,
+	fillable,
+	clickable
+} from 'ember-cli-page-object'
+
+const credentials = {
```
Comment (Rxbsxn) [inline tests/integration/components/add-auction-test.js]: same here

File: tests/integration/components/myprofile-test.js [hunk 11]
```diff
@@ -1,33 +1,92 @@
...
+	clickable
+} from 'ember-cli-page-object'
+
+const credentials = {
+	email: 'integrationtest@herecars.com',
+	password: 'test123',
+	name: 'test name'
+}
+
+const page = create({
```
Comment (Rxbsxn) [inline tests/integration/components/myprofile-test.js]: same here

**Pull request merged**